File: smartmoneyserver/src/main/java/fpt/aptech/server/dto/transaction/report/TransactionTotalDTO.java
package fpt.aptech.server.dto.transaction.report;

import lombok.Builder;

import java.math.BigDecimal;

/**
 * DTO đơn giản để chứa kết quả tính tổng thu và tổng chi.
 * Dùng trong các câu query GROUP BY phức tạp.
 */
@Builder
public record TransactionTotalDTO(
        BigDecimal totalIncome,
        BigDecimal totalExpense
) {
    // Constructor phụ để xử lý trường hợp kết quả từ DB là null
    public TransactionTotalDTO {
        if (totalIncome == null) {
            totalIncome = BigDecimal.ZERO;
        }
        if (totalExpense == null) {
            totalExpense = BigDecimal.ZERO;
        }
    }
}

File: APP MOBILE/smart_money/lib/modules/transaction/models/report/transaction_total_dto.dart
// modules/transaction/models/report/transaction_total_dto.dart
// Map đúng với TransactionTotalDTO.java của Spring Boot
// Kết quả tính tổng thu và tổng chi đơn giản (REPORT)
// Lưu ý: DTO này chủ yếu dùng bên server cho query GROUP BY,
// Flutter có thể không cần dùng trực tiếp nhưng giữ để đồng bộ DTO.

class TransactionTotalDTO {
  final double totalIncome;
  final double totalExpense;

  const TransactionTotalDTO({
    required this.totalIncome,
    required this.totalExpense,
  });

  factory TransactionTotalDTO.fromJson(Map<String, dynamic> json) {
    return TransactionTotalDTO(
      totalIncome: (json['totalIncome'] as num?)?.toDouble() ?? 0.0,
      totalExpense: (json['totalExpense'] as num?)?.toDouble() ?? 0.0,
    );
  }
}

